match/lib/match/nuke.rb

```diff
@@ -15,22 +18,34 @@ class Nuke
     attr_accessor :profiles
     attr_accessor :files
 
+    attr_accessor :storage
+    attr_accessor :encryption
+
     def run(params, type: nil)
       self.params = params
       self.type = type
 
-      params[:workspace] = GitHelper.clone(params[:git_url],
-                                           params[:shallow_clone],
-                                           skip_docs: params[:skip_docs],
-                                           branch: params[:git_branch],
-                                           git_full_name: params[:git_full_name],
-                                           git_user_email: params[:git_user_email],
-                                           clone_branch_directly: params[:clone_branch_directly])
+      self.storage = Storage::Interface.storage_class_for_storage_mode(params[:storage_mode]).new
+      self.storage.configure(git_url: params[:git_url],
```

Comment:
Hmm, reading more of the code, how about all the sub-commands skim the params they need off the top and then return the rest to the storage engine directly? e.g.

```
storage_mode = params.delete(:storage_mode)
readonly = params.delete(:readonly)
# etc.

self.storage = Storage::Interface.storage_class_for_storage_mode(storage_mode).new
self.storage.configure(**params)

# ... etc.
```

That way the parameters can be changed to make sense for the respective storage engine. For example, say I was writing a "simple sftp storage interface" (SSSInterface for short) then things like `git_url` and  `clone_branch_directly` won't make sense, but instead I may want things like `sss_url` or `only_download_subfolder`

Reply:
@milch Are you thinking we should just pass the whole `params` object in to the configure method?